pkg/kv/kvclient/kvcoord/dist_sender.go

```diff
@@ -2056,6 +2056,8 @@ func (ds *DistSender) sendToReplicas(
 			// doesn't have info. Like above, this asks the server to return an
 			// update.
 			ClosedTimestampPolicy: routing.ClosedTimestampPolicy(),
+
+			ExplicitlyRequested: ba.ClientRangeInfo.ExplicitlyRequested,
```

Comment:
Also, please add a unit test that verifies that this works. This could be next to these ones here, which are also a good template:

https://github.com/cockroachdb/cockroach/blob/a095546265339624db9ba0db939c98b596c983c7/pkg/kv/kvclient/kvcoord/dist_sender_server_test.go#L275-L286

Reply:
Test added. I also made the test send to multiple ranges, even though my usage is only unsplitable requests, and added the required logic to a `BatchResponse_Header.combine()` for that to pass as well.